### ISSUE TITLE
Proper Handling of Roman Numerals in Accessible Content

### ISSUE THREAD

**Issue by gregoriopellegrino** [daisy/kb#62]
In many cases, Roman numerals are used as the sole identifier for chapters or sections of a book or other written work. For example, a chapter may be titled "IV" instead of "Chapter 4". However, this poses a challenge for users who rely on assistive technology to access digital content. Screen readers and other assistive devices may interpret "IV" as two separate letters ("I" and "V"), making it difficult for these users to understand the structure of the content.

One possible solution would be to use an aria-label or another accessible attribute to provide an alternative text description of the Roman numeral. For example, a chapter entitled "IV" could have an aria-label of "Chapter 4: Title of Chapter". This would allow assistive technology to correctly announce the number as "Chapter 4" while still preserving the Roman numeral as the visual identifier.

I would like to request guidance on the proper way to handle Roman numerals in accessible content, particularly when used as chapter or section titles. Thank you for your attention to this matter.

**Comment by mattgarrish** [comment]
Roman numerals are problematic wherever they're used. This is one of the many cases for why we need better TTS pronunciation support, as using an alternative label for headings only addresses one specific case.

But given that a usable TTS support language is probably still years off, I guess documenting the use of labels for headings wouldn't hurt as an intermediary solution.

> a chapter entitled "IV" could have an aria-label of "Chapter 4: Title of Chapter".

Not sure what you mean here by adding "Title of chapter". Did you mean the heading is also "IV Title of Chapter"? An accessible label shouldn't introduce information that isn't in the book as that puts an unrealistic expectation on the publisher/remediator to come up with a title that the author never produced. If the only heading is "IV" the label should be "Four". That it's the heading for a chapter will come from the role and heading tag applied.

**Comment by mattgarrish** [comment]
For the record, this only sounds like a helpful practice to suggest, not a requirement that anyone would have to do. Having roman numerals mispronounced, or the letters read out, by AT wouldn't fail content, as they pose the same issues for sighted readers (if you don't know your roman numerals, for example, what does coming across the string "mcmlxxv" in the text mean to you?). AT can spell out the string of characters if the user finds them confusing.

**Comment by gregoriopellegrino** [comment]
I agree. Maybe it can be included as a best-practice? Is using the `abbr` element inappropriate?

**Comment by mattgarrish** [comment]
> Is using the `abbr` element inappropriate?

I wouldn't think so, yes. Roman numerals are more an alternative writing system for numbers; they're not abbreviations or initialisms that need expansion.

**Comment by gregoriopellegrino** [comment]
Here are some considerations about Roman numerals: https://www.w3.org/WAI/WCAG2/supplemental/patterns/o3p06-format-punctuation/

**Comment by clapierre** [comment]
Adding a consideration that came up in discussion with a colleague, relevant to projects like Project Gutenberg where fidelity to the source text is a core requirement.

The aria-label/alternate-text approach (e.g., aria-label="four" on "Chapter IV", or aria-label="the Eighth" on "Henry VIII") solves the pronunciation problem, but it can introduce a worse one: it can collapse a distinction that exists in the source text.

If a book has "Chapter IV" in one place and "Chapter 4" in another — which does happen, e.g. front matter using Roman numerals and body text using Arabic numerals, or numbering conventions that shift between a translator's notes and the main text — normalizing both to the same spoken output ("four") makes them indistinguishable to a screen reader user. A sighted reader can see the notation differs, even without knowing what (if anything) that difference signifies. An AT user loses that information entirely, silently, with no way to know a normalization even happened.

So for text-fidelity-sensitive contexts, I'd suggest the guidance include an explicit option that preserves the notation distinction rather than only normalizing to a value:

aria-label="four, Roman numeral IV" (chapter/section numbers — cardinal) or aria-label="the Eighth, Roman numeral VIII" (personal ordinals — e.g. monarch names) — announces the value and flags that it was a Roman numeral, so it doesn't collide with an Arabic "4" elsewhere in the same document
or leaving Roman numerals unlabeled and letting AT spell them out as-written, relying on document structure (headings, TOC/landmarks) rather than pronunciation to convey chapter identity — arguably the most faithful option, since sighted readers unfamiliar with Roman numerals are in the same position

Worth noting as a caveat regardless of which labeling approach is chosen: aria-label overrides the accessible name entirely, which affects braille output too — so a braille reader would see the label text, not the literal glyphs "IV." There's no single attribute that preserves visual fidelity, braille fidelity, and correct pronunciation all at once; any solution involves a tradeoff, and which one to accept probably depends on the specific numbering context (chapter header vs. inline reference vs. list marker).

**Comment by gregoriopellegrino** [comment]
You're right, keeping the Roman numerals in the label is important, especially because if the element were interactive (such as a link), omitting the numeral from the label would violate the WCAG 2.5.3 "Label in Name" success criterion.

Perhaps the best approach would be to use `aria-description`, which is specified in ARIA 1.3, though it is still in the Working Draft stage.

**Comment by mattgarrish** [comment]
The problem here is that we're trying to bludgeon a TTS issue with ARIA. We really shouldn't be advising people to hack their books for a problem that AT users have always faced. An aural reader will make sense of the character strings in the same way that a visual reader will, and if the AT mispronounces the string they can always step through the characters. What we need is support for technologies like pls and ssml.

The more hacks we suggest here the more that has to be undone later.

**Comment by gregoriopellegrino** [comment]
Yes, you're right.
